Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerDotNetCore.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.PythonTools.LanguageServerClient {
+    internal class PythonLanguageServerDotNetCore : PythonLanguageServer {
+        private readonly IServiceProvider _site;
+
+        private const string ExeName = "Microsoft.Python.LanguageServer.exe";
+        private const string DllName = "Microsoft.Python.LanguageServer.dll";
+
+        public PythonLanguageServerDotNetCore(IServiceProvider site) {
+            _site = site ?? throw new ArgumentNullException(nameof(site));
+        }
+
+#if DEBUG
+        // Since VS is 32-bit process, Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles)
+        // gives us 32-bit program files so can't use that
+        private static string DotNetExeFilePath = Path.Combine(
+            @"C:\Program Files",
+            "dotnet",
+            "dotnet.exe"
+        );
+#endif
+
+        public async override Task<Connection> ActivateAsync() {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var serverFolderPath = GetServerLocation();
+
+            await Task.Yield();
+
+            var info = CreateStartInfo(serverFolderPath);
+
+            var process = new Process {
+                StartInfo = info
+            };
+
+            if (process.Start()) {
+                return new Connection(process.StandardOutput.BaseStream, process.StandardInput.BaseStream);
+            }
+
+            return null;
+        }
+
+        public override object CreateInitializationOptions(string interpreterPath, string interpreterVersion, string rootPath, IEnumerable<string> searchPaths) {
```

Comment:
should we add string checks?

Reply:
searchPaths was the only that could cause a crash, I've added a check for null.